src/providers/ipa/selinux_child.c

```diff
@@ -345,9 +349,20 @@ int main(int argc, const char *argv[])
 
     DEBUG(SSSDBG_TRACE_FUNC, "performing selinux operations\n");
 
-    needs_update = seuser_needs_update(ibuf);
+    /* When using domain_resolution_order the username will always be
+     * fully-qualified, what has been causing some SELinux issues as mappings
+     * for user 'admin' are not applied for 'admin@ipa.example'.
+     *
+     * In order to work this around we can take advantage that selinux_child
+     * queries SSSD since commit 92addd7ba and call getpwnam() in order to get
+     * the username in the correct format. */
+    passwd = getpwnam(ibuf->username);
```

Comment:
Nitpick, but can you add a MINOR_FAILURE here if passwd is NULL telling the admin that getpwnam failed and therefore the SELinux label setting might fail as well?

Reply:
I've updated the patch, thanks for the review!